shell/platform/windows/text_input_plugin_unittest.cc

```diff
@@ -137,5 +142,61 @@ TEST(TextInputPluginTest, VerifyComposingSendStateUpdate) {
   EXPECT_TRUE(sent_message);
 }
 
+TEST(TextInputPluginTest, TextEditingWorksWithDeltaModel) {
+  auto handled_message = CreateResponse(true);
+  auto unhandled_message = CreateResponse(false);
+  int received_scancode = 0;
+
+  TestBinaryMessenger messenger(
+      [&received_scancode, &handled_message, &unhandled_message](
+          const std::string& channel, const uint8_t* message,
+          size_t message_size, BinaryReply reply) {});
+  EmptyTextInputPluginDelegate delegate;
+
+  int redispatch_scancode = 0;
+  TextInputPlugin handler(&messenger, &delegate);
+
+  auto args = std::make_unique<rapidjson::Document>(rapidjson::kArrayType);
+  auto& allocator = args->GetAllocator();
+  args->PushBack(123, allocator);  // client_id
+
+  rapidjson::Value client_config(rapidjson::kObjectType);
+  client_config.AddMember(kEnableDeltaModel, true, allocator);
+
+  args->PushBack(client_config, allocator);
+  auto encoded = JsonMethodCodec::GetInstance().EncodeMethodCall(
+      MethodCall<rapidjson::Document>(kSetClientMethod, std::move(args)));
+
+  EXPECT_TRUE(messenger.SimulateEngineMessage(
+      kChannelName, encoded->data(), encoded->size(),
+      [](const uint8_t* reply, size_t reply_size) {}));
+
+  handler.KeyboardHook(VK_RETURN, 100, WM_KEYDOWN, '\n', false, false);
+  handler.ComposeBeginHook();
+  std::u16string text;
+  text.push_back('\n');
+  handler.ComposeChangeHook(text, 1);
+  handler.ComposeEndHook();
+
+  handler.KeyboardHook(0x4E, 100, WM_KEYDOWN, 'n', false, false);
+  handler.ComposeBeginHook();
+  std::u16string textN;
+  text.push_back('n');
+  handler.ComposeChangeHook(textN, 1);
+  handler.KeyboardHook(0x49, 100, WM_KEYDOWN, 'i', false, false);
+  std::u16string textNi;
+  text.push_back('n');
+  text.push_back('i');
+  handler.ComposeChangeHook(textNi, 2);
+  handler.KeyboardHook(VK_RETURN, 100, WM_KEYDOWN, '\n', false, false);
+  std::u16string textChineseCharacter;
+  text.push_back(u'\u4F60');
+  handler.ComposeChangeHook(textChineseCharacter, 1);
+  handler.ComposeCommitHook();
+  handler.ComposeEndHook();
+
```

Comment:
It would be nice if I could verify the deltas that are generated here, but I wasn't able to do it by mocking TextInputClient, and I also couldn't figure out if it's possible to listen to the message sent to the framework by SendStateUpdateWithDeltas.  If anyone knows of a way, let me know.